src/components/appointments/CreateAppointment/CreateAppointment.stories.js

```diff
@@ -202,6 +202,17 @@ export const HappyFlow = {
       await userEvent.click(button);
     });
 
+    await step('Wait for the declarations to load', async () => {
+      await waitFor(
+        async () =>
+          await expect(
+            await canvas.getByLabelText(
+              /I accept the privacy policy and consent to the processing of my personal data/
+            )
+          )
+      );
+    });
```

Comment:
I think you can use `await canvas.findByLabelText` which will wait for everything to load, but I'm not 100%. So that might help you get rid of the `waitFor` (it should be similar to handling the dynamic product/location list dropdowns).